synapseclient/core/utils.py

```diff
@@ -326,6 +326,56 @@ def is_synapse_id_str(obj):
     return None
 
 
+def bool_or_none(input_value: str) -> typing.Union[bool, None]:
+    """
+    Attempts to convert a string to a bool. Returns None if it fails.
+
+    Args:
+        input_value: The string to convert to a bool
+
+    Returns:
+        The bool or None if the conversion fails
+    """
+    if input_value == "True" or input_value == "true":
```

Comment:
Will we need to worry about 1 and 0?

Nit: At this point in the `if else` we are assuming that `input_value` will most likely be a string, could we do `input_value.lower()` to simplify this just in case we get strings like `TRUE`

Reply:
I think 1 and 0 should rightfully be parsed as an int as that's how pandas would be interpret the value.

I'll add in a case insensitive check here.

Reply:
I updated to allow case insensitivity here